src/ast/passes/semantic_analyser.cpp

```diff
@@ -574,6 +574,55 @@ static const std::map<std::string, call_spec> CALL_SPEC = {
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::string },
           arg_type_spec{ .type=Type::integer, .literal=true } } } },
+  { "strcmp",
+    { .min_args=2,
+      .max_args=2,
+      .discard_ret_warn = true,
+      .arg_types={
+          arg_type_spec{ .type=Type::string },
+          arg_type_spec{ .type=Type::string } } } },
```

Comment:
One of the advantages of the string kfuncs is that the arguments do not need to be in BPF memory. By limiting them to `Type::string`, we effectively loose that advantage. The solution could be to allow both strings and pointers here but I don't know how much we support polymorphic builtin functions. We should discuss this further.